crates/ra_ide/src/inlay_hints.rs

```diff
@@ -1052,4 +1098,124 @@ fn main() {
         "###
         );
     }
+
+    #[test]
+    fn chaining_hints_ignore_comments() {
+        let (analysis, file_id) = single_file(
+            r#"
+            struct A(B);
+            impl A { fn into_b(self) -> B { self.0 } }
+            struct B(C)
+            impl B { fn into_c(self) -> C { self.0 } }
+            struct C;
+
+            fn main() {
+                let c = A(B(C))
+                    .into_b() // This is a comment
+                    .into_c();
```

Comment:
Hmm, I didn't have experience with chaining hints in Intelij, but shouldn't there be a hint for the last method call?
Though this may look weird since it will bias the semicolon and the type may be unnecessary duplicated as there is a variable type hint already.

Reply:
Indeed, the reason I didn't include the final method is because it would be often redundant information. Here, the type hint would show `let c: C = ...`. IntelliJ does the same, I would guess for similar reasons: 
![image](https://user-images.githubusercontent.com/13765376/77483276-164b4400-6e28-11ea-9d3b-b5624cbcc52c.png)